dandi/files.py

```diff
@@ -77,7 +77,7 @@
 
 # TODO -- should come from schema.  This is just a simplistic example for now
 _required_dandiset_metadata_fields = ["identifier", "name", "description"]
-_required_nwb_metadata_fields = ["subject_id"]
+_required_nwb_checks = [check_subject_exists, check_subject_id_exists]
```

Comment:
so is the idea here that we will need to adjust this list of checks to cover only the checks we care about?

could there instead be some kind of annotation of checks in nwbinspector, e.g. with `dandi` tag of some kind, so we could just ask here for all checks with `"dandi"` tag instead of asking for specifically these checks?

asking because hardcoding a list of checks here would require not only a release of nwbinspector whenever we add a new test but also changes to dandi-cli and its release.  Best to have a workflow which would allow for needing only a single library release, e.g. `nwbinspector` and asking people to update it instead of updating both.

Reply:
> so is the idea here that we will need to adjust this list of checks to cover only the checks we care about?

For hard-enforcing this level of validation, yes. Basically what you were originally doing with exact metadata assertions (`subject_id`), but now each assertion is a 

> could there instead be some kind of annotation of checks in nwbinspector, e.g. with dandi tag of some kind, so we could just ask here for all checks with "dandi" tag instead of asking for specifically these checks?

So, we have a [config functionality](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/nwbinspector/internal_configs/dandi.inspector_config.yaml) that alters Importance levels of checks, but still includes all original checks from the base registry. You can tell it to SKIP various checks at whim, though, but for this it might be better to instead have a SELECT operation.

> asking because hardcoding a list of checks here would require not only a release of nwbinspector whenever we add a new test but also changes to dandi-cli and its release. Best to have a workflow which would allow for needing only a single library release, e.g. nwbinspector and asking people to update it instead of updating both.

I see what you mean, here. We will have to think about how best to allow this on our end, then. Will get back soon